src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.Process.cs

```diff
@@ -66,5 +66,49 @@ private static RemoteInvokeHandle RemoteInvoke(MethodInfo method, string[] args,
                 Process.Start(psi) :
                 new Process() { StartInfo = psi }, options);
         }
+
+        /// <summary>A cleanup handle to the Process created for the remote invocation.</summary>
+        public sealed class RemoteInvokeHandle : IDisposable
+        {
+            public RemoteInvokeHandle(Process process, RemoteInvokeOptions options)
+            {
+                Process = process;
+                Options = options;
+            }
+
+            private Process Process { get; set; }
+            public RemoteInvokeOptions Options { get; private set; }
+
+            public void Dispose()
+            {
+                if (Process != null)
+                {
+                    // A bit unorthodox to do throwing operations in a Dispose, but by doing it here we avoid
+                    // needing to do this in every derived test and keep each test much simpler.
+                    try
+                    {
+                        Assert.True(Process.WaitForExit(Options.TimeOut),
+                            $"Timed out after {Options.TimeOut}ms waiting for remote process {Process.Id}");
+
+                        if (Options.CheckExitCode)
+                        {
+                            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                                Assert.Equal(Options.ExpectedExitCode, Process.ExitCode);
+                            else
+                                Assert.Equal(unchecked((sbyte)Options.ExpectedExitCode), unchecked((sbyte)Process.ExitCode));
```

Comment:
Yes that looks right! Thanks for confirming :) Looks good, I just checked your changes and look good.